### ISSUE TITLE
Plugin execution is not triggered on merge commits

### ISSUE THREAD

**Issue by akurkin** [ashwanthkumar/gocd-build-github-pull-requests#71]
hello guys,
consider this situation:
- create **feature branch** from **master**
- make changes, commit and push -- pipeline will be executed automatically
- few PRs went into **master**
- merge **master** into **feature branch** -- this generates a merge commit
- push to remote repo -- it won't be picked up by gocd plugin

Is it known limitation? any way to work around it?

**Comment by gunnarx** [comment]
I just had a PR not trigger the plugin this morning, not understanding why.  After starting the pipeline manually the build status was reported to the PR.

I am not sure if it followed the exact sequence you describe here since I was only playing around to try out the plugin, but I can confirm that the topmost commit in the PR was indeed a merge commit.

**Comment by ashwanthkumar** [comment]
Does it pick up the latest changes when you trigger the pipeline manually?

**Comment by akurkin** [comment]
@ashwanthkumar yes, and it also works if I trigger for specific revision (revision is properly shown) the problem is that it's not triggered automatically

**Comment by ashwanthkumar** [comment]
@akurkin Can you share the plugin logs? If possible enable debug logging on the go server and if you can retry the above steps and share the logs, that would be even great.

**Comment by whiskeysierra** [comment]
We're experiencing the exact same issue: Automatic triggering works, unless it's a merge commit. Unfortunately I don't have access to the log files.

**Comment by AlexanderYastrebov** [comment]
Confirm, autobuild does not work after merge commit push

**Comment by ashwanthkumar** [comment]
Thanks for the issue report. I'll have a look at it and keep you guys
posted.

On Tue, May 24, 2016 at 8:50 PM, Alexander Yastrebov <
notifications@github.com> wrote:

> Confirm, autobuild does not work after merge commit push
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ashwanthkumar/gocd-build-github-pull-requests/issues/71#issuecomment-221305892>
>



-- 

Ashwanth Kumar / ashwanthkumar.in

**Comment by AlexanderYastrebov** [comment]
Hello. I have created pull request to fix it. Actual fix is just removing break in cycle.
#74

**Comment by akurkin** [comment]
that's awesome. Looking forward to seeing this issue fixed!

**Comment by ashwanthkumar** [comment]
@AlexanderYastrebov Thank you so much for getting this done. I'll verify and merge them. 

Meanwhile @akurkin @whiskeysierra - can you guys also verify the changes from your side?

**Comment by akurkin** [comment]
Sure, happy to do it. Is there a jar file available or some instructions on how to build it myself? (sorry, not well versed in java world)

**Comment by ashwanthkumar** [comment]
@akurkin Please check out [github-pr-poller-1.2.4-merge-commit-fix.jar.zip](https://github.com/ashwanthkumar/gocd-build-github-pull-requests/files/282455/github-pr-poller-1.2.4-merge-commit-fix.jar.zip)

PS: Github doesn't support attaching JAR files on the issues. So had to compress into into a ZIP :(

**Comment by ashwanthkumar** [comment]
Here's the Git Feature Branch plugin (with fixes from #74), if you're directly using it from Git repos without Github.

[git-fb-poller-1.2.4-merge-commit-fix.jar.zip](https://github.com/ashwanthkumar/gocd-build-github-pull-requests/files/282465/git-fb-poller-1.2.4-merge-commit-fix.jar.zip)

**Comment by whiskeysierra** [comment]
> @akurkin Please check out github-pr-poller-1.2.4-merge-commit-fix.jar.zip

I tried it, but a merge commit still doesn't trigger a build for my PR:

See my attached screenshots. Last build was an hour ago, but the last changes show my merge from 7 minutes ago.

![screenshot from 2016-06-23 11-07-33](https://cloud.githubusercontent.com/assets/429981/16297846/dcb37998-3932-11e6-9b30-a39b5bdb0a7f.png)
![screenshot from 2016-06-23 11-08-13](https://cloud.githubusercontent.com/assets/429981/16297847/dcb4064c-3932-11e6-8850-1436c545b663.png)

**Comment by decoomanj** [comment]
Same issue here, any news on this?

**Comment by decoomanj** [comment]
It seems that when the modified-material list is empty, the pipeline is NOT triggered. This happens when e.g. GitHub does a merge (master into a branch, for example). The change is detected, but nothing happens because the modified-material-list is empty. 

I have made a plain work-around by inserted a fake-change in the list to trigger the pipeline. This is probably not the way to go, but I didn't really see another solution since the Java-Git library doesn't give me extra information.

Take a look here for the [change](https://github.com/solarisBank/gocd-build-github-pull-requests/commit/e150310eb11a65010613874f376f9a99df1cfec9).

**Comment by ashwanthkumar** [comment]
I just made a release with the changes on PR at ashwanthkumar/git-cmd#2 

https://github.com/ashwanthkumar/gocd-build-github-pull-requests/releases/tag/v1.3.0-RC2

Please check and let me know if that fixes the problem. 

cc: @juhasipo

**Comment by whiskeysierra** [comment]
> Please check and let me know if that fixes the problem.

Not for us, no.

**Comment by Josaphat** [comment]
I'm still seeing the issue as well.

**Comment by whiskeysierra** [comment]
Actually I just realized that we're still using the old version of the plugin. I have to check again, next time a merge appears.

**Comment by whiskeysierra** [comment]
Nope, doesn't work.

**Comment by ashwanthkumar** [comment]
Okay, I finally got some time to look into this issue and it seems that for merge commits the modifiedFiles list seems to be empty, even after the changes from https://github.com/ashwanthkumar/git-cmd/pull/2/ that doesn't seem to help for cases wherein there're only new files added via merge commits. Added an ugly patch to force trigger the commits and it seems to be working. The fix is part of https://github.com/ashwanthkumar/gocd-build-github-pull-requests/pull/94.

**Comment by qlibin** [comment]
@ashwanthkumar It seems like the issue is back again in v1.3.3

I tried to deploy gocd server `v17.2.0.2` with `github-pr-poller-1.3.3.jar` - the issue's back.

Then I deployed gocd server `v17.2.0.2` with `github-pr-poller-1.3.1.jar` - the issue's gone.

**Comment by ashwanthkumar** [comment]
@qlibin Oops looks like a regression, most likely as part of [this](https://github.com/ashwanthkumar/gocd-build-github-pull-requests/commit/b7a631acf291cd81108dee230bd76fa9606818ca). Will fix it but this time with tests around that part. Thanks for reporting.

**Comment by GordonGuenther** [comment]
This issue also seems to applies to the stash-pr-poller. Would you be so kind to fix it there as well?

**Comment by jinayvora1990** [comment]
This issue also exists on git-fb-poller-1.3.4.jar. Will this be fixed too ?

**Comment by amithalef** [comment]
We are still facing this issue @ashwanthkumar, is there any updates on this?
